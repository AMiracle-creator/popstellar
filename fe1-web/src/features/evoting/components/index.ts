export { default as ElectionListItem, ElectionEventType } from './ElectionListItem';
