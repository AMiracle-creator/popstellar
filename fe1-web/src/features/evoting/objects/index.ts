export * from './Configuration';
export * from './Election';
export * from './Feature';
