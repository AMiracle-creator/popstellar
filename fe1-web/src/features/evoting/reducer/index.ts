export * from './ElectionKeyReducer';
export { default as electionKeyReducer } from './ElectionKeyReducer';
