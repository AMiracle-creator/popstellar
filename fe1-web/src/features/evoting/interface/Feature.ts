import { LaoEventsParamList } from 'core/navigation/typing/LaoEventsParamList';
import { NavigationScreen } from 'core/navigation/typing/Screen';
import { Hash, PublicKey } from 'core/objects';

export namespace EvotingFeature {
  export interface Lao {
    id: Hash;
    organizer: PublicKey;
    witnesses: PublicKey[];
  }

  export interface EventState {
    eventType: string;
    id: string;
    idAlias?: string;

    start: number;
    end?: number;
  }

  export interface LaoEventScreen extends NavigationScreen {
    id: keyof LaoEventsParamList;
  }
}
