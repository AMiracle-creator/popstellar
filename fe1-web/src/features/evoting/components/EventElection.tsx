import PropTypes from 'prop-types';
import React, { FunctionComponent, useMemo } from 'react';
import { useSelector } from 'react-redux';

import STRINGS from 'resources/strings';

import { Election, ElectionStatus } from '../objects';
import { makeElectionSelector } from '../reducer';
import ElectionNotStarted from './ElectionNotStarted';
import ElectionOpened from './ElectionOpened';
import ElectionResult from './ElectionResult';
import ElectionTerminated from './ElectionTerminated';

/**
 * Component used to display a Election event in the LAO event list
 */

const EventElection = (props: IPropTypes) => {
  const { eventId: electionId, isOrganizer } = props;

  const selectElection = useMemo(() => makeElectionSelector(electionId), [electionId]);
  const election = useSelector(selectElection);

  if (!election) {
    throw new Error(`Could not find an election with id ${electionId}`);
  }

  const questions = useMemo(
    () => election.questions.map((q) => ({ title: q.question, data: q.ballot_options })),
    [election.questions],
  );

  switch (election.electionStatus) {
    case ElectionStatus.NOT_STARTED:
      return (
        <ElectionNotStarted election={election} questions={questions} isOrganizer={isOrganizer} />
      );
    case ElectionStatus.OPENED:
      return <ElectionOpened election={election} questions={questions} isOrganizer={isOrganizer} />;
    case ElectionStatus.TERMINATED:
      return <ElectionTerminated election={election} />;
    case ElectionStatus.RESULT:
      return <ElectionResult election={election} />;
    default:
      console.warn('Election Status was undefined in Election display', election);
      return null;
  }
};

const propTypes = {
  eventId: PropTypes.string.isRequired,
  isOrganizer: PropTypes.bool,
};
EventElection.propTypes = propTypes;
EventElection.defaultProps = {
  isOrganizer: false,
};

type IPropTypes = PropTypes.InferProps<typeof propTypes>;

export default EventElection;

export const ElectionEventType = {
  eventType: Election.EVENT_TYPE,
  navigationNames: {
    createEvent: STRINGS.organizer_navigation_creation_election,
  },
  Component: EventElection as FunctionComponent<{
    eventId: string;
    isOrganizer: boolean | null | undefined;
  }>,
};
