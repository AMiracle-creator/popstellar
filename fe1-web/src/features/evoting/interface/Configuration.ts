import React from 'react';
import { AnyAction, Reducer } from 'redux';

import { MessageRegistry } from 'core/network/jsonrpc/messages';
import { Hash } from 'core/objects';
import FeatureInterface from 'core/objects/FeatureInterface';

import { ElectionReducerState, ELECTION_REDUCER_PATH } from '../reducer/ElectionReducer';
import { EvotingFeature } from './Feature';

export const EVOTING_FEATURE_IDENTIFIER = 'evoting';

export interface EvotingConfiguration {
  // objects
  messageRegistry: MessageRegistry;

  /* LAO related functions */

  /**
   * Returns the currently active lao. Should be used outside react components
   * @returns The current lao
   */
  getCurrentLao: () => EvotingFeature.Lao;

  /**
   * Returns the currently active lao. Should be used inside react components
   * @returns The current lao
   */
  useCurrentLao: () => EvotingFeature.Lao;

  /**
   * Returns the currently active lao id. Should be used inside react components
   * @returns The current lao id
   */
  useCurrentLaoId: () => Hash | undefined;

  /* Event related functions */

  /**
   * Creates a redux action for adding an event to the event store
   * @param laoId - The lao id where to add the event
   * @param event - The event
   * @returns A redux action causing the state change
   */
  addEvent: (laoId: Hash | string, event: EvotingFeature.EventState) => AnyAction;

  /**
   * Creates a redux action for update the stored event state
   * @param event - The event
   * @returns A redux action causing the state change
   */
  updateEvent: (event: EvotingFeature.EventState) => AnyAction;

  /**
   * Given the redux state and an event id, this function looks in the active
   * lao for an event with a matching id, creates an instance of the corresponding type
   * and returns it
   * @param id - The id of the event
   * @returns The event or undefined if none was found
   */
  getEventById: (id: Hash) => EvotingFeature.EventState | undefined;
}

/**
 * The type of the context that is provided to react evoting components
 */
export type EvotingReactContext = Pick<
  EvotingConfiguration,
  /* lao */
  | 'useCurrentLao'
  | 'useCurrentLaoId'
  /* events */
  | 'getEventById'
  | 'addEvent'
  | 'updateEvent'
>;

/**
 * The interface the evoting feature exposes
 */
export interface EvotingInterface extends FeatureInterface {
  screens: {
    CreateElection: React.ComponentType<any>;
  };

  eventTypes: EventType[];

  context: EvotingReactContext;

  reducers: {
    [ELECTION_REDUCER_PATH]: Reducer<ElectionReducerState>;
  };
}

interface EventType {
  eventType: string;
  navigationNames: {
    createEvent: string;
  };
  Component: React.ComponentType<{
    eventId: string;
    isOrganizer: boolean | null | undefined;
  }>;
}
