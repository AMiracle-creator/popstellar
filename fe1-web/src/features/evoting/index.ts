import { ElectionEventType } from './components';
import { EvotingConfiguration, EvotingInterface, EVOTING_FEATURE_IDENTIFIER } from './interface';
import { configureNetwork } from './network';
import { electionReducer } from './reducer';
import { CreateElectionScreen, ViewSingleElectionScreen } from './screens';

/**
 * Configures the e-voting feature
 *
 * @param config - A evoting configuration object
 * @returns
 */
export const configure = (config: EvotingConfiguration): EvotingInterface => {
  const { useCurrentLao, useCurrentLaoId, getEventById, addEvent, updateEvent } = config;
  // configure message registry to correctly handle incoming messages
  configureNetwork(config);
  // return the interface that is exposed by the evoting feature
  return {
    /* this context will be used to pass the properties to react components */
    identifier: EVOTING_FEATURE_IDENTIFIER,
    laoEventScreens: [CreateElectionScreen, ViewSingleElectionScreen],
    eventTypes: [ElectionEventType],
    context: {
      /* lao */
      useCurrentLao,
      useCurrentLaoId,
      /* event */
      getEventById,
      addEvent,
      updateEvent,
    },
    reducers: {
      ...electionReducer,
    },
  };
};
