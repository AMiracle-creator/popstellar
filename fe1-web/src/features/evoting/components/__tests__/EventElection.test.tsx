import { render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';
import { combineReducers, createStore } from 'redux';

import {
  mockLao,
  mockLaoIdHash,
  messageRegistryInstance,
  mockReduxAction,
  mockLaoId,
} from '__tests__/utils';
import FeatureContext from 'core/contexts/FeatureContext';
import { addEvent } from 'features/events/reducer';
import {
  mockElectionNotStarted,
  mockElectionOpened,
  mockElectionResults,
  mockElectionTerminated,
} from 'features/evoting/__tests__/utils';
import { EVOTING_FEATURE_IDENTIFIER } from 'features/evoting/interface';
import { Election, ElectionStatus } from 'features/evoting/objects';
import { addElection, electionReducer, updateElection } from 'features/evoting/reducer';

import EventElection from '../EventElection';

const undefinedElection = Election.fromState({
  ...mockElectionNotStarted.toState(),
  electionStatus: 'undefined' as ElectionStatus,
});

const mockStore = createStore(combineReducers({ ...electionReducer }));
mockStore.dispatch(
  addEvent(mockLaoId, {
    eventType: Election.EVENT_TYPE,
    id: mockElectionNotStarted.id.valueOf(),
    start: mockElectionNotStarted.start.valueOf(),
    end: mockElectionNotStarted.end.valueOf(),
  }),
);
mockStore.dispatch(addElection(mockElectionNotStarted.toState()));

const contextValue = {
  [EVOTING_FEATURE_IDENTIFIER]: {
    useCurrentLao: () => mockLao,
    useCurrentLaoId: () => mockLaoIdHash,
    addEvent: () => mockReduxAction,
    updateEvent: () => mockReduxAction,
    getEventById: () => undefined,
    messageRegistry: messageRegistryInstance,
    onConfirmEventCreation: () => undefined,
  },
};

describe('EventElection', () => {
  describe('Not started election', () => {
    beforeAll(() => {
      mockStore.dispatch(updateElection(mockElectionNotStarted.toState()));
    });

    it('renders correctly for an organizer', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionNotStarted.id.valueOf()} isOrganizer />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('renders correctly for an attendee', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionNotStarted.id.valueOf()} />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });
  });

  describe('Opened election', () => {
    beforeAll(() => {
      mockStore.dispatch(updateElection(mockElectionOpened.toState()));
    });

    it('renders correctly for an organizer', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionOpened.id.valueOf()} isOrganizer />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('renders correctly for an attendee', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionOpened.id.valueOf()} />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });
  });

  describe('Terminated election where the results are not yet available', () => {
    beforeAll(() => {
      mockStore.dispatch(updateElection(mockElectionTerminated.toState()));
    });

    it('renders correctly for an organizer', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionTerminated.id.valueOf()} isOrganizer />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('renders correctly for an attendee', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionTerminated.id.valueOf()} />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });
  });

  describe('Finished election where the results are available', () => {
    beforeAll(() => {
      mockStore.dispatch(updateElection(mockElectionResults.toState()));
    });

    it('renders correctly for an organizer', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={mockElectionResults.id.valueOf()} isOrganizer />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('renders correctly for an attendee', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection
              eventId={mockElectionResults.id.valueOf()}
              start={mockElectionResults.start.valueOf()}
              end={mockElectionResults.end.valueOf()}
            />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });
  });

  describe('Undefined election status', () => {
    beforeAll(() => {
      mockStore.dispatch(updateElection(undefinedElection.toState()));
    });

    it('renders null for an organizer', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={undefinedElection.id.valueOf()} isOrganizer />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('renders null for an attendee', () => {
      const component = render(
        <Provider store={mockStore}>
          <FeatureContext.Provider value={contextValue}>
            <EventElection eventId={undefinedElection.id.valueOf()} />
          </FeatureContext.Provider>
        </Provider>,
      ).toJSON();
      expect(component).toMatchSnapshot();
    });
  });
});
