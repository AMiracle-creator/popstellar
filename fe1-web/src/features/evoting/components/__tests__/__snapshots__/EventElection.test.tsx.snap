// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventElection Not started election renders correctly for an attendee 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            "Ballot Option 1",
            "Ballot Option 2",
          ],
          "title": "Mock Question 1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Mock Question 1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginHorizontal": 20,
              "textAlign": "center",
            }
          }
        >
          • Ballot Option 1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginHorizontal": 20,
              "textAlign": "center",
            }
          }
        >
          • Ballot Option 2
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>,
]
`;

exports[`EventElection Not started election renders correctly for an organizer 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <RCTScrollView
    data={
      Array [
        Object {
          "data": Array [
            "Ballot Option 1",
            "Ballot Option 2",
          ],
          "title": "Mock Question 1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={
      Array [
        0,
      ]
    }
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 20,
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Mock Question 1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginHorizontal": 20,
              "textAlign": "center",
            }
          }
        >
          • Ballot Option 1
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginHorizontal": 20,
              "textAlign": "center",
            }
          }
        >
          • Ballot Option 2
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>,
]
`;

exports[`EventElection Running election renders correctly for an attendee 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "zIndex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "marginHorizontal": 10,
          "textAlign": "center",
        }
      }
    >
      Mock Question 1
    </Text>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "zIndex": 3,
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fafafa",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 1,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="checkBoxBallot Option 1"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#bfbfbf",
                  "fontSize": 24,
                },
                Object {
                  "minWidth": 24,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
          >
            Ballot Option 1
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fafafa",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 1,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="checkBoxBallot Option 2"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#bfbfbf",
                  "fontSize": 24,
                },
                Object {
                  "minWidth": 24,
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
          >
            Ballot Option 2
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginHorizontal": 50,
        "marginVertical": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Cast Vote
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#52c41a",
          "borderColor": "#fff",
          "borderRadius": 9,
          "borderWidth": 0.5,
          "height": 18,
          "justifyContent": "center",
          "minWidth": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 12,
            "paddingHorizontal": 4,
          }
        }
      >
        0
      </Text>
    </View>
  </View>,
]
`;

exports[`EventElection Running election renders correctly for an organizer 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
        "zIndex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
          "marginHorizontal": 10,
          "textAlign": "center",
        }
      }
    >
      Mock Question 1
    </Text>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "zIndex": 3,
        }
      }
    >
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fafafa",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 1,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="checkBoxBallot Option 1"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
          >
            Ballot Option 1
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="checkbox"
        accessibilityState={
          Object {
            "checked": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fafafa",
            "borderColor": "#ededed",
            "borderRadius": 3,
            "borderWidth": 1,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "opacity": 1,
            "padding": 10,
          }
        }
        testID="checkBoxBallot Option 2"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "color": "#43484d",
                "fontWeight": "bold",
                "marginLeft": 10,
                "marginRight": 10,
              }
            }
            testID="checkboxTitle"
          >
            Ballot Option 2
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginHorizontal": 50,
        "marginVertical": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Cast Vote
        </Text>
      </View>
    </View>
  </View>,
  <View
    style={Object {}}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#52c41a",
          "borderColor": "#fff",
          "borderRadius": 9,
          "borderWidth": 0.5,
          "height": 18,
          "justifyContent": "center",
          "minWidth": 18,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 12,
            "paddingHorizontal": 4,
          }
        }
      >
        0
      </Text>
    </View>
  </View>,
  <View
    style={
      Object {
        "marginHorizontal": 50,
        "marginVertical": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Terminate Election / Tally Votes
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`EventElection Terminated election where the results are available renders correctly for an attendee 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Election Terminated
  </Text>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Waiting for result
  </Text>,
]
`;

exports[`EventElection Terminated election where the results are available renders correctly for an organizer 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Election Terminated
  </Text>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Waiting for result
  </Text>,
]
`;

exports[`EventElection Terminated election where the results are not yet available renders correctly for an attendee 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Election Terminated
  </Text>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Waiting for result
  </Text>,
]
`;

exports[`EventElection Terminated election where the results are not yet available renders correctly for an organizer 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "marginHorizontal": 10,
        }
      }
    >
      Start: 
      <time
        dateTime="2020-12-31T23:00:00.000Z"
        title="2020-12-31 23:00"
      >
        1 year ago
      </time>
      

      End: 
      <time
        dateTime="2021-01-01T23:00:00.000Z"
        title="2021-01-01 23:00"
      >
        1 year ago
      </time>
    </Text>
  </View>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Election Terminated
  </Text>,
  <Text
    style={
      Object {
        "fontSize": 25,
        "marginHorizontal": 10,
        "textAlign": "center",
      }
    }
  >
    Waiting for result
  </Text>,
]
`;
